Story/vspc 171

glend17 opened the pull request:
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 
(you can copy the ticket if it hasn't changed) 

https://diging.atlassian.net/browse/VSPC-171
  
## Anything else the reviewer needs to know?
I have added a new content block type called Space Block, which has the same properties as the content block with an additional parameter for linking to a space. I have allowed the selection of a space for the Staff User from a drop down

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java:
```diff
@@ -361,4 +428,20 @@ private void updateContentOrder(String slideId, Integer contentOrder) {
+            }
+            
+        }
+        return selectedSpaceForSpaceBlock;
```
what is this method for? why does it retrieve space blocks again? (also if you. need to check the type of an object you should use `instanceof`).

glend17 replied:
Removed this method as it was not needed

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/DeleteSpaceBlockController.java:
```diff
@@ -0,0 +1,38 @@
+
+        } catch (BlockDoesNotExistException e) {
+            logger.warn("Text Id does not exist, bad request.", e);
+            return new ResponseEntity<String>(HttpStatus.BAD_REQUEST);
```
if a block doesn't exist, there should be a 404 returned

glend17 replied:
Done

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/SlideController.java:
```diff
@@ -36,20 +46,31 @@
+        List<ISpace> spaces = new ArrayList<ISpace>();
+        spaceRepo.findAll().forEach(s -> {
+            spaces.add(s);
+        });
```
why are the spaces added to a new list? doesn't `findAll` return a list?

glend17 replied:
Done

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/SlideController.java:
```diff
@@ -36,20 +46,31 @@
             model.addAttribute("choices", ((IBranchingPoint)slide).getChoices());
-        }           
+        }      
+        System.out.println("the map is"+ selectedSpaceForSpaceBlock.get("CON000000007"));
```
Really? you even checked the box saying "I do not have any sysout statements in my code or commented out code that isn’t needed anymore" ;)

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -73,7 +74,25 @@
+            		var blockId = $(this).closest("div").attr("id");
+            		console.log('block id is'+ blockId)
+            		var selectedSpace = selectedSpaceForSpaceBlock[blockId]
+            		console.log('selected space is '+selectedSpace)
```
remove

jdamerow replied:
all console.log

glend17 replied:
Removed

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -73,7 +74,25 @@
+                    $(".open").css('display', 'none');
+                    $(".open").css('overflow-y', 'scroll');
+                    $(".open").css('height', '400px');
+                }
```
fix indentations in above block

glend17 replied:
The indentation appears correctly in eclipse i am not sure why it appears like this here.
<img width="645" alt="Screen Shot 2022-02-23 at 12 35 57 PM" src="https://user-images.githubusercontent.com/22446190/155394667-44cac677-bb1a-4ffa-abc2-16349bed10cb.png">

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/SlideController.java:
```diff
@@ -36,20 +46,31 @@
         List<IContentBlock> slideContents = contentBlockManager.getAllContentBlocks(id);
+        HashMap<String, String> selectedSpaceForSpaceBlock = contentBlockManager.findSelectedSpaceForSpaceBlock(slideContents);
         model.addAttribute("slideContents", slideContents);
+        model.addAttribute("selectedSpaceForSpaceBlock", selectedSpaceForSpaceBlock);
```
what are these needed for? each space block should have a space property with the linked space.

glend17 replied:
Removed this, initially had included this to make the user know which is the current selected space for a space content block, but I have figured a way around this.

jdamerow commented:
Resolve conflicts please

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/DeleteSpaceBlockController.java:
```diff
@@ -0,0 +1,41 @@
+    @RequestMapping(value = "/staff/module/{moduleId}/slide/{id}/space/{blockId}", method = RequestMethod.DELETE)
+    public ResponseEntity<String> deleteSpaceBlock(@PathVariable("id") String slideId,@PathVariable("blockId") String blockId) throws IOException {
+        logger.info("block id is {}", blockId);
+        logger.info("module id is", slideId);
```
remove

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditSpaceBlockController.java:
```diff
@@ -0,0 +1,48 @@
+            @RequestParam("spaceBlockId") String blockId, @PathVariable("moduleId") String moduleId,
+            @RequestParam("spaceBlockTitle") String spaceBlockTitle,
+            @RequestParam("spaceId") String spaceId) throws IOException {
+        logger.info("spaceId is {}", spaceId);
```
remove

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/SlideController.java:
```diff
@@ -36,11 +46,15 @@
+    public String listSlides(@PathVariable("id") String id, @PathVariable("moduleId") String moduleId, Model model) throws JsonProcessingException {
 
         ISlide slide = slideManager.getSlide(id);
+        model.addAttribute("spaces", spaceRepo.findAll());
```
this does not seem like a very good way to do that. what if there are 100 spaces? Better search spaces via an ajax call when a new slide is being created.

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -87,7 +90,23 @@
+                    currentSelectedSpace = $("div.open p:eq(1)").text();
+                	defaultValue = description;
+            		var editSpaceAlertHTML = document.getElementById('#editSpaceAlert')
+            		var blockId = $(this).closest("div").attr("id");
```
indentation

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -595,13 +640,43 @@
+                				$(".error").delay(4000).slideUp(500, function(){
+                				    $(".error").empty();
+                				});
+                			}
```
why is the error clause further indented than the success block?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -713,6 +789,45 @@
+                    var text = markDown.value();
+                	var title = document.getElementById('spaceTitle').value;
+                	var selectedSpace = document.getElementById('spaceName');
+                	var spaceId = selectedSpace.options[selectedSpace.selectedIndex].value;
```
indentation

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -713,6 +789,45 @@
+                        	else {
+                            	var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>We are sorry but something went wrong. Please try again later.</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+                            	$('.error').append(alert);
+                        	}
```
indentation

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -837,7 +975,45 @@
+                       		else {
+                           		var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>We are sorry but something went wrong. Please try again later.</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+                           		$('.error').append(alert);
+                       		}
```
indentation

jdamerow reviewed (changes requested):
just look at the files on GitHub after you open a PR and you should easily see where the indentation is off.

glend17 commented:
Closing the pr as there are new indentation issues with latest changes on slide.html

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditSpaceBlockController.java:
```diff
@@ -0,0 +1,57 @@
+        return new ResponseEntity<String>(HttpStatus.OK);
+    }
+    
+    @RequestMapping(value = "/staff/module/slide/spaceblock/{id}", method = RequestMethod.GET)
```
should be `/staff/module/{moduleId}/slide/{id}/space/{id}`

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/ListSpacesController.java:
```diff
@@ -33,4 +40,10 @@ public String listSpaces(Model model) {
         return "staff/spaces/spacelist";
     }
+    
+    @RequestMapping("/staff/module/slide/listspaces")
```
should be `/staff/spaces/search`

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/ListSpacesController.java:
```diff
@@ -33,4 +40,10 @@ public String listSpaces(Model model) {
+    
+    @RequestMapping("/staff/module/slide/listspaces")
+    public ResponseEntity<List<ISpace>> listSpacesForSlideContentBlock(){
+        List<ISpace> spaces = spaceManager.getAllSpaces();
```
this still loads all spaces. Instead it should take a query string and then search spaces for the query string (let's just go with searching in the space title for now).

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -70,6 +72,62 @@
+            	    newOption.text = ele.name;
+            	    newOption.value = ele.id;
+            	    element.appendChild(newOption);
+            	});
```
since we're using jquery, let's be consistent and use it here as well.

jdamerow replied:
and everywhere else in your JS code

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -70,6 +72,62 @@
                 $(this).removeClass("hova");
             }
             
+            function formatSpaceListResultsInEditSpaceBlock(result){
```
how about `createSpaceDropdown`?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -70,6 +72,62 @@
+            	
+            }
+            
+            function removeAllSelectOptionsInEditBlock(elementId){
```
how about `removeOptions`?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -70,6 +72,62 @@
+            
+            function populateSpacesInEditSpaceBlock(currentSelectedSpace){
+            	$.ajax({
+                    	url: "[(@{'/staff/module/slide/listspaces/'})]"+"?"+[[${_csrf.parameterName}]]+'='+[[${_csrf.token}]],
```
csrf token should be in header

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -70,6 +72,62 @@
+            	element.innerHTML="";
+            }
+            
+            function populateSpacesInEditSpaceBlock(currentSelectedSpace){
```
how about `findSpaces`?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -70,6 +72,62 @@
+            	
+            }
+            
+            function getCurrentSelectedSpaceForSpaceBlock(spaceBlockId){
```
`getSelectedSpace`

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -70,6 +72,62 @@
+            
+            function getCurrentSelectedSpaceForSpaceBlock(spaceBlockId){
+            	$.ajax({
+        		url: "[(@{'/staff/module/slide/spaceblock/'})]" + spaceBlockId +"?"+[[${_csrf.parameterName}]]+'='+[[${_csrf.token}]],
```
csrf in header

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -503,6 +578,44 @@
+            		    type: 'GET',
+            		    success: function(result) {
+            		        formatSpaceListResults(result);
+            		        $(".EasyMDEContainer").remove();
```
is that necessary?

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -84,6 +88,15 @@ public void test_deleteTextBlockById_forNonExistentId() throws BlockDoesNotExist
+        when(contentBlockRepository.findById("notARealId")).thenReturn(contentBlockOptional);
+        Mockito.doThrow(EmptyResultDataAccessException.class).when(spaceBlockRepo).deleteById(spaceBlockId);
+        managerToTest.deleteSpaceBlockById(spaceBlockId,"slideId_1");
+    }
```
what about adding/editing/deleting space blocks?

jdamerow replied:
?

jdamerow replied:
what about editing space blocks?

jdamerow replied:
where?

ajay3568yadav replied:
<img width="1047" alt="Screenshot 2024-05-08 at 15 33 27" src="https://github.com/diging/virtual-spaces-2.0/assets/89062459/30931a86-ccc1-45db-a500-d51e92a1037d">

jdamerow replied:
I got so confused since the comment was so old 😅

glend17 commented:
Added Ajax calls using select2, dynamically searches for spaces in the drop down, with pagination included. 
Added a unit test for creating space block, update space block i think does not need a unit test as it directly updates the form with the repo.
Added csrf into headers and addressed other comments on path and javascript function name changes, with ajax calls i have removed some functions that are not necessary

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -70,6 +72,25 @@
+    		type: 'GET',
+    		success: function(result) {
+    				let selectedSpace = document.getElementById("selectedSpace");
+    				selectedSpace.innerHTML = result;
```
let's use jquery like the rest

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceBlockController.java:
```diff
@@ -0,0 +1,43 @@
+    @Autowired
+    private ISpaceManager spaceManager;
+    
+    @RequestMapping(value = "/staff/module/{moduleId}/slide/{slideId}/SpaceBlockContent", method = RequestMethod.POST)
```
should be `.../slide/{slideId}/space`

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceBlockController.java:
```diff
@@ -0,0 +1,39 @@
+    @Autowired
+    private IContentBlockManager contentBlockManager;
+    
+    @RequestMapping(value = "/staff/module/{moduleId}/slide/{id}/spaceblock/{blockId}", method = RequestMethod.GET)
```
`.../slide/{id}/space/{blockId}`

glend17 commented:
addressed review comments

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java:
```diff
@@ -201,6 +223,49 @@ public void deleteTextBlockById(String blockId, String slideId) throws BlockDoes
     }
+    
+    @Override
+    public ISpace getCurrentSpaceForSpaceBlock(String blockId, String slideId) throws BlockDoesNotExistException {
```
just noticing this now, why is this a method? Why not just get the spaceblock in the controller and then call `getSpace`? That's what it's therefore after all.

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java:
```diff
@@ -267,6 +332,11 @@ public void deleteChoiceBlockById(String blockId, String slideId) throws BlockDo
     }
+    
+    @Override
+    public void updateSpaceBlock(SpaceBlock spaceBlock) {
```
let's call this `saveSpaceBlock` (since I just talked to Avirup about something like this).

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceBlockController.java:
```diff
@@ -0,0 +1,43 @@
+    private ISpaceManager spaceManager;
+    
+    @RequestMapping(value = "/staff/module/{moduleId}/slide/{slideId}/space", method = RequestMethod.POST)
+    ResponseEntity<String> addSpaceBlock(@RequestParam("title") String title,@RequestParam("spaceId") String spaceId, @PathVariable("slideId") String slideId,
```
should be public

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/ListSpacesController.java:
```diff
@@ -1,12 +1,22 @@
+import edu.asu.diging.vspace.core.model.ISpace;
 import edu.asu.diging.vspace.core.services.IExhibitionManager;
 import edu.asu.diging.vspace.core.services.ISpaceManager;
 
```
does this file need these imports?

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/ListSpacesInContentBlockController.java:
```diff
@@ -0,0 +1,45 @@
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+
+@Controller
+public class ListSpacesInContentBlockController {
```
why not `SearchSpacesController`?

jdamerow replied:
also, should be in the api package

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/ListSpacesInContentBlockController.java:
```diff
@@ -0,0 +1,45 @@
+    private ISpaceManager spaceManager;
+    
+    @RequestMapping("/staff/spaces/search")
+    public ResponseEntity<String> listSpacesForSlideContentBlock(@RequestParam(value = "term", required = false) String searchTerm){
```
`search(...)`

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/SlideController.java:
```diff
@@ -36,9 +46,12 @@
     private IContentBlockManager contentBlockManager;
+    
+    @Autowired
+    private SpaceRepository spaceRepo;
```
is this needed and all the imports?

jdamerow replied:
?

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceBlockController.java:
```diff
@@ -0,0 +1,39 @@
+import edu.asu.diging.vspace.core.exception.BlockDoesNotExistException;
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.model.ISpaceBlock;
+import edu.asu.diging.vspace.core.model.impl.SpaceBlock;
```
is this needed?

jdamerow reviewed (changes requested):
run "organize imports" on your files.

glend17 commented:
Addressed review comments and Merged develop to resolve conflicts

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java:
```diff
@@ -297,4 +304,83 @@ public Iterable<Space> addIncomingLinkInfoToSpaces(Iterable<Space> spaces) {
+    @Override
+    public List<ISpace> getSpaces(int pageNo, String sortedBy, String order) {
+        Sort sortingParameters = getSortingParameters(sortedBy, order);
+        pageNo = validatePageNumber(pageNo);
```
The two methods `validatePageNumber` and `getTotalPages` only seem to be used from here (so they should be private) but also I'm not sure if you even need them. I think if you try to get a page number that's higher than the number of pages it simply returns no result. But the page object you get back has the number of pages in it, so in that case you can simply make another call to the db. This way in most cases you only make one DB call except when `pageNo` is greater than max pages. The way it's implemented right now there are always two db calls.

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/forms/SpaceContentBlockForm.java:
```diff
@@ -0,0 +1,20 @@
+public class SpaceContentBlockForm {
+    
+    String title;
+    String spaceName;
```
should be private

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -70,6 +72,25 @@
             
+            function getSelectedSpace(blockId){
+            	$.ajax({
+                	url: "[(@{'/staff/module/'+${module.id}+'/slide/'+${slide.id}+'/space/'})]"+ blockId + "?" +[[${_csrf.parameterName}]]+'='+[[${_csrf.token}]],
```
get requests don't need a csrf token

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -503,6 +539,66 @@
+                		return space.name;
+                	}
+                	var $space = $(
+                			'<span>' + space.name + '</span>'
```
indent

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -503,6 +539,66 @@
+                }
+                
+                function formatSpaceSelection(space) {
+                	if (!space.id) {
```
indent

jdamerow replied:
?

glend17 commented:
Addressed review PR comments

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java:
```diff
@@ -387,4 +455,9 @@ private void updateContentOrder(String slideId, Integer contentOrder) {
     }
+
+    @Override
+    public void updateSpaceBlock(SpaceBlock spaceBlock) {
```
this method should only take `ISpaceBlock`

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java:
```diff
@@ -267,6 +321,11 @@ public void deleteChoiceBlockById(String blockId, String slideId) throws BlockDo
     }
+    
+    @Override
+    public void saveSpaceBlock(SpaceBlock spaceBlock) {
```
this method should only take ISpaceBlock

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/SearchSpacesController.java:
```diff
@@ -0,0 +1,45 @@
+    private ISpaceManager spaceManager;
+    
+    @RequestMapping("/staff/spaces/search")
+    public ResponseEntity<String> searchSpacesForSlideContentBlock(@RequestParam(value = "term", required = false) String searchTerm){
```
why not just `search`?

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/SearchSpacesController.java:
```diff
@@ -0,0 +1,45 @@
+        ObjectMapper mapper = new ObjectMapper();
+        ArrayNode idArray = mapper.createArrayNode();
+        for (ISpace space : spaces) {
+            ObjectNode imageNode = mapper.createObjectNode();
```
why is this called `imageNode`?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -87,13 +108,28 @@
+                    let spaceId = e.target.closest('div').id;
+                    $(e.target).closest('.SpaceDivStaff').addClass("open");
+                    var description = $("div.open p:eq(0)").text();
+                    defaultValue = description;
```
what's the point of this?

glend17 replied:
When cancel button is clicked from edit space block or text block, the boxes are re-updated with the title that was before editing it, to save making a call to the backend. 
This is there even in line 102 which was from a previous ticket and I have used the same logic for this ticket as well

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -837,7 +1058,48 @@
-                });    
+                });   
+                $("#submitSpaceBlock").on("click", function(e){
+                	var token =  $('input[name="_csrf"]').attr('value');
```
this seems unnecessary, you should be able to get the token through thymeleaf.

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -84,6 +102,47 @@ public void test_deleteTextBlockById_forNonExistentId() throws BlockDoesNotExist
+    }
+    
+    @Test
+    public void test_editSpaceBlock_valid() throws BlockDoesNotExistException {
```
there doesn't seem to be a `editSpaceBlock` method

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -84,6 +102,47 @@ public void test_deleteTextBlockById_forNonExistentId() throws BlockDoesNotExist
+        ISpaceBlock createdBlock = managerToTest.createSpaceBlock(slideId, title, contentOrder, space);
+        Assert.assertEquals(createdBlock.getContentOrder(), contentOrder);
+    }
 
```
what about the `getSpaceBlock` method?

jdamerow replied:
?

bnayak4 replied:
I am writing test cases and increased the code coverage from 38% to 72%

glend17 commented:
Addressed review comments and re raised PR.

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
- there needs to be a title for the space drop down and the elements need to be properly aligned.
![image](https://user-images.githubusercontent.com/8881141/169338495-c7ad945b-d9ae-4e34-949c-9bf574840793.png)
-  The tooltip for the button and the title of the modal should be "Add link to space" not "Add space block"
- After adding a new space block, the block does not show which space is linked from it.

bnayak4 commented:
@jdamerow can you explain the editing space block comment

jdamerow commented:
which one?

jdamerow commented:
There are test failures.

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/SlideController.java:
```diff
@@ -49,7 +52,7 @@ public String listSlides(@PathVariable("id") String id, @PathVariable("moduleId"
+        }
         return "staff/modules/slides/slide";
     }
 
```
?

bnayak4 replied:
@jdamerow We need to declare JSONprocessingException that is the only change in this file.

jdamerow replied:
where is that exception being thrown? and why is `SpaceRepository` imported, it doesn't seem to be used?

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -82,8 +101,65 @@ public void test_deleteTextBlockById_forNonExistentId() throws BlockDoesNotExist
+        assertNull(retrievedSpaceBlock);
+        
+      
+    }
```
please reread our test guidelines. Test methods don't follow our naming convention.

bnayak4 replied:
@jdamerow are you refering to this method - test_getSpaceBlock_forNonExistentId() ? If yes I will change it to test_getSpaceBlock_missingId()

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -82,9 +149,73 @@ public void test_deleteTextBlockById_forNonExistentId() throws BlockDoesNotExist
+    }
+
+    @Test
+    public void test_deleteSpaceBlock_success() throws BlockDoesNotExistException {
```
method is not following our naming convention

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -82,9 +149,73 @@ public void test_deleteTextBlockById_forNonExistentId() throws BlockDoesNotExist
+        Mockito.when(spaceBlockRepo.save((SpaceBlock) spaceBlock)).thenReturn((SpaceBlock) spaceBlock);
+        Mockito.when(spaceBlockFactory.createSpaceBlock(slide, title, (ISpace) space)).thenReturn(spaceBlock);
+        ISpaceBlock createdBlock = managerToTest.createSpaceBlock(slideId, title, contentOrder, space);
+        Assert.assertEquals(createdBlock.getContentOrder(), contentOrder);
```
what about the other properties of the space block? They should be asserted as well.

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+        when(slideRepo.findById("notARealId")).thenReturn(Optional.empty());
+
+        List<IContentBlock> contentBlock = managerToTest.getAllContentBlocks(slideId);
+        assertTrue(!contentBlock.isEmpty());
```
this needs to check if the correct content blocks are returned.

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+        contentBlocks.add(secondContentBlock);
+        slide.setContents(contentBlocks);
+        when(slideManager.getSlide(slideId)).thenReturn(slide);
+        when(slideRepo.findById("notARealId")).thenReturn(Optional.empty());
```
is this needed?

jdamerow replied:
?

ajay3568yadav replied:
yes cuz when we rearrange the content order changes and this test , tests that

jdamerow replied:
when is `slideRepo.findById("notARealId")` called in this test?

ajay3568yadav replied:
removed that, cuz it did seem unnecessary.

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+        String blockId = "realBlockId";
+        String slideId = "realSlideId";
+        when(contentBlockRepository.findById(blockId)).thenReturn(Optional.of(contentBlock));
+        managerToTest.deleteChoiceBlockById(blockId, slideId);
```
there is nothing asserted here. when does the test succeed?

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+    public void test_deleteChoiceBlockById_nullBlockId() throws BlockDoesNotExistException {
+        String blockId = null;
+        String slideId = "realSlideId";
+        managerToTest.deleteChoiceBlockById(blockId, slideId);
```
see above

jdamerow replied:
there seem to be more test cases for `deleteChoiceBlockById` further down. They should all be together.

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+        when(slideManager.getSlide(slideId)).thenReturn(slide);
+        when(imageFactory.createImage(fileName, fileName)).thenReturn(slideContentImage);
+        when(imageRepo.save((VSImage) slideContentImage)).thenReturn((VSImage) slideContentImage);
+        managerToTest.updateImageBlock(imageBlock, image, fileName);
```
see above

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+        slideContentImage.setHeight(700);
+        slideContentImage.setWidth(1300);
+        IImageBlock imageBlock = new ImageBlock();
+        managerToTest.updateImageBlock(imageBlock, slideContentImage);
```
see above

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+    }
+
+    @Test
+    public void test_updateImageBlock_success1() throws ImageCouldNotBeStoredException {
```
what scenario is `success1`? needs better name.

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+    }
+
+    @Test
+    public void test_updateImageBlock_success2() throws ImageCouldNotBeStoredException {
```
see above

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+        when(imageBlockRepo.save((ImageBlock) imageBlock)).thenReturn((ImageBlock) imageBlock);
+        Mockito.when(imageRepo.save((VSImage) slideContentImage)).thenReturn((VSImage) slideContentImage);
+        Mockito.when(imageBlockFactory.createImageBlock(slide, slideContentImage)).thenReturn(imageBlock);
+        managerToTest.createImageBlock(slideId, image, fileName, contentOrder);
```
there are no assertions. when does this test succeed?

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+    }
+
+    @Test
+    public void test_createImageBlock2_success() throws ImageCouldNotBeStoredException, FileStorageException {
```
method name does not follow naming convention

bnayak4 replied:
Many methods in that class has got same name, how to address them?

jdamerow replied:
the name should express what the outcome of the test is, so not just "success" but "successSomeCondition" expressing what the tested condition is.

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+        when(imageBlockRepo.save((ImageBlock) imageBlock)).thenReturn((ImageBlock) imageBlock);
+        Mockito.when(imageBlockFactory.createImageBlock(slide, slideContentImage)).thenReturn(imageBlock);
+        CreationReturnValue returnValue = managerToTest.createImageBlock(slideId, slideContentImage, contentOrder);
+        assertNotNull(returnValue);
```
so whenever the method returns anything, this test succeeds? the assertions need to test that the method actually returns what was expected.

jdamerow replied:
line 519 is now not needed. if `returnValue` were null, line 517 would throw a null pointer

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+        Mockito.when(textBlockRepo.save((TextBlock) textBlock)).thenReturn((TextBlock) textBlock);
+        ITextBlock createdBlock = managerToTest.createTextBlock(titleString, text, contentOrder);
+
+        Assert.assertEquals(createdBlock.getContentOrder(), contentOrder);
```
what about the other properties of the text block

jdamerow reviewed (changes requested):
Please reread out test naming guidelines.

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -1,48 +1,85 @@
+}
+</style>
+<script th:inline="javascript">
+=======
```
???

jdamerow replied:
and it looks like there are two `<head>` now

jdamerow replied:
and some other duplications

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -161,13 +217,28 @@
+		    		if(imgBlockID != '') {
+		        		$("#uploadImage").data('value', imgBlockID); // sets image ID value
+		        		$("#addImgAlert").show();
+		    		}
```
indentation

jdamerow replied:
?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -1269,456 +1568,953 @@
+$("#videoTitle").val(title); $("#saveVideo").data('value', vidId);
+$("#addVideoAlert").show(); } });
+</script>
+</head>
```
please fix this file

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
Layout is off (dropdown is squeezed down in the bottom left corner), I can't see the field I should type the title into, and the tooltip overlays the modal.
![image](https://user-images.githubusercontent.com/8881141/235482986-e07e4cc5-c3ef-43ed-8c86-310fe3155985.png)
![image](https://user-images.githubusercontent.com/8881141/235482998-974b208e-12db-4296-b6f9-b8aa9ea6634c.png)

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
There are compilation errors.

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java:
```diff
@@ -242,10 +264,12 @@ public CreationReturnValue createVideoBlock(String slideId, byte[] video, Long s
     }
 
-    private IVSVideo storeVideo(byte[] video, Long size, String fileName, String url, String title)
+    public IVSVideo storeVideo(byte[] video, Long size, String fileName, String url, String title)
```
is this method ever called outside this class (or is it supposed to be)? if not, it should not be public

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java:
```diff
@@ -140,7 +162,7 @@ private IVSVideo saveVideoWithUrl(String url, String title) {
     }
 
-    private IVSVideo saveVideo(byte[] video, Long size, String filename, String title) {
+    public IVSVideo saveVideo(byte[] video, Long size, String filename, String title) {
```
is this method ever called outside this class (or is it supposed to be)? if not, it should not be public

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -214,9 +218,79 @@ public void test_deleteTextBlockById_forNonExistentId() throws BlockDoesNotExist
+
+    @Test
+    public void test_updateSpaceBlock_success() {
+        String upadtedSpaceId = "spaceId1";
```
"updated" is misspelled

jdamerow replied:
?

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
when I create a new space block, then edit it, the space drop down is empty. If I save it like that, the space link is set to null. It needs to retain the current space. Ideally, the drop down list should show the current space.

also, on curation side, the block needs to indicate to which space a block is linked.

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java:
```diff
@@ -297,4 +304,59 @@ public Iterable<Space> addIncomingLinkInfoToSpaces(Iterable<Space> spaces) {
+     *                page is returned
+     * @return list of images in the requested pageNo and requested order.
+     */
+
```
remove empty line

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java:
```diff
@@ -297,4 +304,59 @@ public Iterable<Space> addIncomingLinkInfoToSpaces(Iterable<Space> spaces) {
+        if(pageNo < 1) {
+            pageNo = 1;
+        }
+        Pageable sortByRequestedField = PageRequest.of(pageNo - 1, pageSize, sortingParameters);
```
variable names should usually be nouns or maybe adjectives but not actions (that's for method names)

jdamerow replied:
It's fine to call this just `pagable` (that's what it represents after all)

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/SearchSpacesController.java:
```diff
@@ -0,0 +1,45 @@
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+
+@Controller
+public class SearchSpacesController {
```
Is this needed? Can't you use the `StaffSearchSpaceController`?

ajay3568yadav replied:
they are actually being used for different tasks

jdamerow replied:
What's different about them?

jdamerow replied:
?

ajay3568yadav replied:
one is mapped to "/staff/spaces/search" and the other is to "/staff/search/space"

jdamerow replied:
That doesn't make sense. What's already there should be reused. It's fine if the search searches more than the name.

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -1,55 +1,55 @@
+    $('.close').on('click', function () {
+        $('.modal').hide();
+    })
+})
```
formatting should stay the same

jdamerow replied:
?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -144,6 +89,25 @@
+                	url: "[(@{'/staff/module/'+${module.id}+'/slide/'+${slide.id}+'/space/'})]"+ blockId,
+    		type: 'GET',
+    		success: function(result) {
+    			//$("#selectedSpace").text('');
```
remove if not needed

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -161,13 +125,28 @@
+                    let spaceId = e.target.closest('div').id;
+                    $(e.target).closest('.SpaceDivStaff').addClass("open");
+                    var description = $("div.open p:eq(0)").text();
+                    defaultValue = description;
```
merge with line 131

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -924,36 +858,16 @@
-                    $("#uploadImage").data('value', '');
+                	$('#chooseFromLocal').trigger( "click" );
+                	// Initialize selected image ID to blank, on clicking cancel button
+                	$("#uploadImage").data('value', '');
```
indentation

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -924,36 +858,16 @@
-                    $("#selectedImage").empty();
-                    $('#imageId').prop('selectedIndex',0).trigger( "change" );
+                	$("#selectedImage").empty();
+                	$('#imageId').prop('selectedIndex',0).trigger( "change" );
```
same

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/config/ExhibitionLanguageConfigTest.java:
```diff
@@ -75,7 +77,12 @@ public void test_init_withNullPropertySource() {
+        Map< String, String> map2 = new HashMap();
+        map1.put("key2", "value2") ;
+        
+        List<Map> languageList = Arrays.asList(map1, map2);
```
why was this changed?

ajay3568yadav replied:
for flexibility(mutable maps)

jdamerow replied:
why? what flexibility is needed here?

ajay3568yadav replied:
I did not make any changes in the test files, must be Prachi, baishali, in their previous commits. these are not from my commits

ajay3568yadav replied:
should I change it back?

jdamerow replied:
Why did you say then that it was for flexibility then? Yes, please change it back.

ajay3568yadav replied:
fixed

jdamerow reviewed (changes requested):
there are unaddressed comments

jdamerow reviewed (changes requested):
There are unaddressed comments.

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -1,55 +1,50 @@
+            	$('.close').on('click', function () {
+                	$('.modal').hide();
+            	})
+        	})
```
indentation

ajay3568yadav replied:
addressed review comments where needed and made sure to address every review comment; please let me know if there is any comment I missed. I fixed the SpaceController one and almost all the indentation ones, and that why reopening the PR.

jdamerow replied:
indentation still

jdamerow reviewed (changes requested):
there are unaddressed comments; you have to look at all the files, even the ones that are collapsed because they are so big!

jdamerow reviewed (changes requested):
unaddressed review comments (I did not comment on them all, please due your due diligence).

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSearchSpaceController.java:
```diff
@@ -33,6 +40,26 @@ public ResponseEntity<StaffSearchSpaceResults> searchInVspace(
     }
+    
+    @RequestMapping("/staff/spaces/search")
+    public ResponseEntity<String> search(@RequestParam(value = "term", required = false) String search){
```
frontend should make a call to `searchInVspace`. No reason to implement a second type of search.

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/UpdateContentOrderController.java:
```diff
@@ -23,17 +30,24 @@ public class UpdateContentOrderController {
-        } catch (BlockDoesNotExistException e) {
-            logger.warn("Block Id does not exist, bad request.", e);
-            return new ResponseEntity<List<ContentBlock>>(HttpStatus.BAD_REQUEST);
+            System.out.println(contentBlockString);
```
?

ajay3568yadav replied:
i was trying to fix the content ordering, but realised that it is not a part of this ticket so I left that part

jdamerow replied:
so why is there a system.out.println in there?

ajay3568yadav replied:
That was for checking the contents of the JSON stringified object, removed the print statement now.

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/UpdateContentOrderController.java:
```diff
@@ -23,17 +30,24 @@ public class UpdateContentOrderController {
+            System.out.println(contentBlockString);
+            contentBlockList = objectMapper.readValue(contentBlockString, new TypeReference<List<ContentBlock>>(){});
+            contentBlockManager.updateContentOrder(contentBlockList,slide);
+        } catch (Exception e) {
```
why was `BlockDoesNotExistException` changed to `Exception`?

ajay3568yadav replied:
because it was not a BlockDoesNotExistException, it was because of unserializazble object

jdamerow replied:
you need to catch the exceptions that are thrown, not just `Exception`. If another one is thrown besides `BlockDoesNotExistException`, then that needs to be caught as well.

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -223,29 +196,41 @@
-            			$(imageblock[0]).mouseenter(onMouseEnter).mouseleave(onMouseLeave).dblclick(onDoubleClick);
-            		}
+
+            	    var fileInput = document.getElementById('file'); // Assuming you have a file input element with id 'fileInput'
```
what does this comment mean?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -223,29 +196,41 @@
+            	    var url = "[(@{'/staff/module/'+${module.id}+'/slide/'+${slide.id}+'/image/?'+${_csrf.parameterName}+'='+${_csrf.token}})]";
+            	    if (fileInput.files.length === 0) {
+            	    	$('#chooseFromLocal').trigger( "click" );
+                    	// Initialize selected image ID to blank, on clicking cancel button
```
?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -223,29 +196,41 @@
+                			imageblock = createImageBlockWithExistingImage($('#selectedImage').find('img').attr('src'),image.width);  
+                			$('#slideSpace').append(imageblock); 
+                			$(imageblock[0]).mouseenter(onMouseEnter).mouseleave(onMouseLeave).dblclick(onDoubleClick);
+                		}
```
indentation of all of the above

ajay3568yadav replied:
Reindented everything according to the 4 tab space indentation convention, which was out of place in even in the develop branch's slide.html file

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -819,13 +759,47 @@
             		});
             	});
+                	
+            		// ---------- Delete Space Block -------------
```
indentation of this whole block

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -1269,456 +1219,484 @@
-    </body>
+		</div>
+	</div>
+</body>
```
why is all the video related code deleted from this file?

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/config/ExhibitionLanguageConfigTest.java:
```diff
@@ -12,6 +12,8 @@
 import java.util.List;
 import java.util.Map;
 
+import javax.print.DocFlavor.STRING;
```
?

ajay3568yadav commented on vspace/src/main/java/edu/asu/diging/vspace/core/services/IContentBlockManager.java:
```diff
@@ -57,8 +61,36 @@ IChoiceBlock createChoiceBlock(String slideId, List<String> selectedChoices, Int
+     * 
+     * @param contentBlockList - The list contains the blocks and the updated
+     *                         content order corresponding to each blocks.
+     */
```
this description for the update content order function got displaced while making some changes, it is now mentioned right above the correct function that it describes.

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSearchSpaceController.java:
```diff
@@ -34,15 +41,6 @@ public ResponseEntity<StaffSearchSpaceResults> searchInVspace(
-     * 
-     * @param spacePagenum current page number sent as request parameter in the URL.
-     * @param searchTerm   This is the search string which is being searched.
-     */
```
is there a reason this is being deleted?

jdamerow replied:
?

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/UpdateContentOrderController.java:
```diff
@@ -23,17 +33,23 @@ public class UpdateContentOrderController {
-            return new ResponseEntity<List<ContentBlock>>(HttpStatus.BAD_REQUEST);
+            contentBlockList = objectMapper.readValue(contentBlockString, new TypeReference<List<ContentBlock>>(){});
+            contentBlockManager.updateContentOrder(contentBlockList,slide);
+        } catch (InaccessibleObjectException e) {
```
where is that being thrown?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -1,1466 +1,1747 @@
             var contentCount = [[${#lists.size(slideContents)}]];
             var markDown = null;
             var defaultValue;
+            var slideContents = [[${slideContents}]];
```
where is that being used?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
There seem to be a lot of changes in this file that are actually not doing anything which makes it really hard to review; please undo any changes that are only line indentation, etc.

jdamerow replied:
this still seems to be the case

jdamerow commented:
also, please adhere to our github guidelines

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -143,6 +151,23 @@
+	            });
+            }
             
             function onDoubleClick(e){
```
This is just a terrible way of doing this. There should be a different handler for each type of block. The handlers can be added depending on the classes they have instead of doing a string comparing mega if (as it is now). Line 1412 would then add a different function that only handles space link blocks.

jdamerow replied:
And image blocks also get their own function.

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSearchSpaceController.java:
```diff
@@ -22,29 +26,33 @@ public class StaffSearchSpaceController {
     @Autowired
     private IStaffSearchManager staffSearchManager;
-
-    @RequestMapping(value = "/staff/search/space")
```
what about the search page that is using this controller?

ajay3568yadav replied:
this is not used anywhere, there is a StaffSearch controller that is rendering the spaces, on being searched

jdamerow replied:
What about here: https://github.com/diging/virtual-spaces-2.0/blob/89ac3e069050cad0d3f9eb19414b765ca2362211/vspace/src/main/webapp/WEB-INF/views/staff/search/staffSearch.html#L68

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -689,6 +728,65 @@
+		            '<span>' + space.name + '</span>'
+		            );
+		            return $image;
+	            }
```
how is this function different form `formatSpace`?

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -214,9 +218,80 @@ public void test_deleteTextBlockById_forNonExistentId() throws BlockDoesNotExist
+    }
+
+    @Test
+    public void test_updateSpaceBlock_success() {
```
needs to follow our naming conventions

ajay3568yadav replied:
this method is testing this function: managerToTest.saveSpaceBlock(spaceBlock); so do you mean it should be called test_saveSpaceBlock_success? if so... then there is already this function, but its saving a space block and this one is updating it, so what should I call this method?

jdamerow replied:
I'm sure this is a question you can figure out yourself... ;)

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -301,6 +368,256 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
+        assertEquals(returnedContentBlock.get(0).getId(), contentBlockIdString);
+        assertEquals(returnedContentBlock.size(), contentBlocks.size());
+        assertEquals(returnedContentBlock.get(0).getContentOrder(), Integer.valueOf(3));
+        assertTrue(!returnedContentBlock.isEmpty());
```
if the above asserts are true, then this one will be too (or if this one returns false, then line 387 would have already thrown an exception)

jdamerow commented on vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java:
```diff
@@ -59,38 +74,51 @@ public class ContentBlockManagerTest {
 
+    @Mock
+
+    private SlideRepository slideRepo;
```
where is that used?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -143,34 +150,77 @@
+                
             }
+            
+            function onDoubleClick(e){
```
what is this function for? it only seems to be used once?

ajay3568yadav commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -689,6 +739,55 @@
+		            ajax: {
+			            url: function (params) {
+				            var spaceURL = [[@{|/staff/spaces/search|}]];
+				            return spaceURL;
```
to dynamically generate URLs in case we have user parameters(params)

jdamerow replied:
It's not though, it just returns `spaceURL` which is always the same.

ajay3568yadav commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -1208,6 +1448,47 @@
+		            var selectedSpace = document.getElementById('editSpaceName');
+		            var spaceId = selectedSpace.options[selectedSpace.selectedIndex].value;
+		            var formData = new FormData();
+		            var text = markDown.value();
```
to provide an editor field to edit the name/title of the block

jdamerow replied:
but that does not need any styling, does it? so just an input field should be enough.

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSearchSpaceController.java:
```diff
@@ -43,8 +43,11 @@ public ResponseEntity<StaffSearchSpaceResults> searchInVspace(
      * @param spacePagenum current page number sent as request parameter in the URL.
      * @param searchTerm   This is the search string which is being searched.
      */
+    
```
there should be no empty line

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSearchSpaceController.java:
it doesn't look there are any changes in there that should be kept

jdamerow replied:
?

ajay3568yadav replied:
did not understand what you mean however this method is called in staffSearch.html

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSpacesSearchController.java:
```diff
@@ -0,0 +1,47 @@
+    private ISpaceManager spaceManager;
+
+    @RequestMapping("/staff/spaces/search")
+    public ResponseEntity < String > search(@RequestParam(value = "term", required = false) String search) {
```
there should be no blanks between `<` and `String` and `>` (everywhere here). please follow our code style

jdamerow replied:
?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -143,33 +150,68 @@
+		            	$("#selectedSpace").text(result.name);
+		            },
+		            error: function(data) {
+				            var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>We are sorry but something went wrong. Please try to delete again later.</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
```
there is nothing deleted here, is there? the message is wrong.

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -143,34 +150,77 @@
+	            //remove card border
+	            $(".open").css('display', 'none');
+	            $(".open").css('overflow-y', 'scroll');
+	            $(".open").css('height', '400px');
```
?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -689,6 +731,49 @@
+		            '<span>' + space.name + '</span>'
+		            );
+		            return $space;
+	            }
```
indentation

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -732,6 +817,14 @@
+		            $(alert).attr( 'value', $(this).siblings('input').val());
+		            $('.modal-footer').append(alert);
+	            });
+
```
indentation

ajay3568yadav replied:
this looks right? its just new code

jdamerow replied:
![image](https://github.com/user-attachments/assets/8c279359-41a5-4bb3-9857-1d544c171d9f)

ajay3568yadav replied:
<img width="821" alt="Screenshot 2024-09-13 at 12 53 01" src="https://github.com/user-attachments/assets/999b96d8-c641-4cf3-b49f-f287d5095b22">
this is how it looks on my side, i don't see anything

jdamerow replied:
you probably have tabs and spaces differently than the rest

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -761,13 +854,32 @@
+	            function removeAllSelectOptions(elementId){
+		            var element = document.getElementById(elementId);
+		            element.innerHTML="";
+	            }
```
indentation

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -761,13 +854,32 @@
+		            $("#confirmDeleteImageAlert").find('input').remove();
+		            $("#confirmDeleteChoiceBlockAlert").find('input').remove();
+		            $("#confirmDeleteSpaceAlert").hide();
+	            });
```
indentation

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -819,13 +945,46 @@
             		e.preventDefault();
             		$("#confirmDeleteTextAlert").find('input').remove();
             		$("#confirmDeleteChoiceBlockAlert").find('input').remove();
+		            $("#confirmDeleteSpaceAlert").find('input').remove();
```
?

ajay3568yadav replied:
The indentation looks right, and a new code line has been added.

jdamerow replied:
![image](https://github.com/user-attachments/assets/0209b488-747d-42cc-85f8-071f40b8bbd2)

ajay3568yadav replied:
i checked tabs and everything, this looks right on my side as well as here on the comments side on github, I blame github for this
<img width="602" alt="Screenshot 2024-09-25 at 12 18 52" src="https://github.com/user-attachments/assets/9a5c9ad7-cc0b-4b7b-b100-05f947142ecc">

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -892,6 +1051,7 @@
             	    e.preventDefault();
             	    $("#confirmDeleteTextAlert").find('input').remove();
             	    $("#confirmDeleteImageAlert").find('input').remove();
+		            $("#confirmDeleteSpaceAlert").find('input').remove();
```
?

ajay3568yadav replied:
The indentation looks right, and a new code line has been added for deleting space

jdamerow replied:
![image](https://github.com/user-attachments/assets/ee2b7161-1a53-4fb5-9345-fbba5f0438de)

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -991,6 +1137,45 @@
+			            formData.append('title', title);
+			            formData.append('spaceId', spaceId);
+			            $.ajax({
+				            url: "[(@{'/staff/module/'+${module.id}+'/slide/'+${slide.id}+'/space?' +${_csrf.parameterName}+'='+${_csrf.token}})]", 
```
csrf token should not be added to url but in header

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -1168,10 +1373,45 @@
-            
+                $('.textDivStaff').mouseenter(onMouseEnter).mouseleave(onMouseLeave).dblclick(textBlockDoubleClick);
+                $('.vidDiv').mouseenter(onMouseEnter).mouseleave(onMouseLeave);
+	            $('.SpaceDivStaff').mouseenter(onMouseEnter).mouseleave(onMouseLeave).dblclick(spaceBlockDoubleClick);
```
?

ajay3568yadav replied:
It looks right, just new methods added and instead of double-click, DoubleClick methods specific to button types are created

jdamerow replied:
![image](https://github.com/user-attachments/assets/acf2ec23-fda8-4030-87fd-a3d9a450d20e)

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -1208,12 +1428,49 @@
+		            formData.append('spaceBlockTitle', text);
+		            $("#editSpaceAlert").hide();
+		            $.ajax({
+			            url: "[(@{'/staff/module/'+${module.id}+'/slide/'+${slide.id}+'/space/edit?'+${_csrf.parameterName}+'='+${_csrf.token}})]",
```
csrf should be in header

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -1494,15 +1797,46 @@ <h6>
+                            </div>
+                            <div class="modal-body">
+                                <h6>
+                                    <small>Choose from existing Spaces </small>
```
?

jdamerow replied:
?

ajay3568yadav replied:
that is not happening anymore, null space is not being assigned to any space block

jdamerow replied:
Point being, this header is incorrect, since you can't choose from non-existing spaces, it should just say "Choose a Space"

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
- in list, the selected space needs to be shown
![image](https://github.com/user-attachments/assets/c1ac81e8-965c-46dc-ab91-06fc3b11d11d)
- padding is missing
![image](https://github.com/user-attachments/assets/06f75537-ab84-4fe7-a478-73769f755769)
- close button needs to fit our layout
![image](https://github.com/user-attachments/assets/33552d74-e353-4b39-9185-a10c1339e023)

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceBlockController.java:
```diff
@@ -0,0 +1,33 @@
+    private IContentBlockManager contentBlockManager;
+    
+    @RequestMapping(value = "/staff/module/{moduleId}/slide/{id}/space/{blockId}", method = RequestMethod.GET)
+    public ResponseEntity<ISpaceBlock> getSpaceBlockSpace(@PathVariable("id") String slideId,
```
this is returning a space block, right? not the space of a space block? so the method should be called `getSpaceBlock`

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
there are a bunch of indentation issues in this file

jdamerow reviewed (changes requested):
slide.html has a bunch of unaddressed comments

jdamerow commented:
It's really nothing big that's missing. Some indentation, a method name, some other small things, especially (but not only) in slide.html. Once those are addressed, this should be good to go!

jdamerow commented on vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSpacesSearchController.java:
```diff
@@ -0,0 +1,47 @@
+
+    @RequestMapping("/staff/spaces/search")
+    public ResponseEntity <String> search(@RequestParam(value = "term", required = false) String search) {
+        List < ISpace > spaces = null;
```
not our code style

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -361,6 +406,9 @@
 					data: formData,
+					headers:{
+            			'X-CSRF-Token':token
+            		},
```
?

ajay3568yadav replied:
you said that the CSFR token should not be directly used in the URL but should be passed through header, thats why I passed the token through header.

jdamerow replied:
this is about indentation

jdamerow reviewed (changes requested):
there are unaddressed comments and still issues with formatting

ajay3568yadav commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -649,14 +702,18 @@
                 });
                 
                 $("#submitTitle").click(function() {
+                	var token = [[${_csrf.token}]];
```
you said that CSFR token should not go into the URL link directly but through headers, thats why I put the token through header

ajay3568yadav replied:
you said that the CSFR token should not be directly used in the URL but should be passed through header, thats why I passed the token through header.

jdamerow replied:
same, indentation

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -1494,15 +1862,46 @@ <h6>
+                                    <small>Choose from existing Spaces </small>
+                                </h6>
+                                <select id="spaceName" name="spacesNames" class="form-control-xs target" style="height: 50px; width: 200px;">
+                                    <option selected value="">Choose from existing</option>
```
same here

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
When editing a space block, the space is not select in the drop down. The drop down has whatever was last selected.

Also, not sure what's going on but I have a space block from previous tests, and the slide page doesn't load for some reason every once in a while, and I can't delete the space block, and the button to add a new one doesn't work.

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
current space should be selected in dropdown

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
Some slides don't seem to work: https://diging-dev.asu.edu/vspace-dev/staff/module/MOD000000002/slide/SLI000000021
```
Caused by: org.springframework.expression.spel.SpelEvaluationException: EL1007E: Property or field 'name' cannot be found on null
	at org.springframework.expression.spel.ast.PropertyOrFieldReference.readProperty(PropertyOrFieldReference.java:213)
	at org.springframework.expression.spel.ast.PropertyOrFieldReference.getValueInternal(PropertyOrFieldReference.java:104)
	at org.springframework.expression.spel.ast.PropertyOrFieldReference.access$000(PropertyOrFieldReference.java:51)
	at org.springframework.expression.spel.ast.PropertyOrFieldReference$AccessorLValue.getValue(PropertyOrFieldReference.java:406)
	at org.springframework.expression.spel.ast.CompoundExpression.getValueInternal(CompoundExpression.java:92)
	at org.springframework.expression.spel.ast.SpelNodeImpl.getValue(SpelNodeImpl.java:112)
	at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:337)
	at org.thymeleaf.spring5.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:265)
	... 130 more

```

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
Make it so, Jenkins.

jdamerow commented:
Make it so, Jenkins.

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -143,33 +152,104 @@
+                    url: "[(@{'/staff/module/'+${module.id}+'/slide/'+${slide.id}+'/space/'})]" + blockId,
+                    type: 'GET',
+                    success: function(result) {
+                    	if (result != null && result.space != null && result.space.name != null){
```
I don't think `result.space.name != null` is needed here? If the space name is null, it shouldn't prevent the space block from being created. However, if there is no title, there should be some dummy text (in the space selection list as well as when displaying the block/link), e.g. "<space has to title>"

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -143,33 +152,104 @@
+                            currentSpaceName = result.space.name;
+                            $("#selectedSpace").text(currentSpaceName);
+                            $("#editSpaceBlock").text(result.title);
+                            currentSpaceId = result.space.id;
```
put this line together with line 162

jdamerow replied:
and honestly, not sure if you even need extra variables here. You can just use `result.space.name` and `result.space.id` everywhere.

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -143,33 +152,104 @@
-                	}
-            	
-                }
+            async function spaceBlockDoubleClick(e) {
```
why is this `async`?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -143,33 +152,104 @@
+                                $("#editSpaceName").append(newOption).trigger('change');
+                            } else {
+                                $("#editSpaceName").val(currentSpaceId).trigger('change');
+                            }
```
i'm so confused, what is this if block for?

ajay3568yadav replied:
This if block checks whether a specific option (with a value equal to currentSpaceId) already exists in the #editSpaceName dropdown menu. If the option is missing, a new one is created with currentSpaceName as the display text and currentSpaceId as the value.

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -761,14 +890,37 @@
+					$("#confirmDeleteTextAlert").find('input').remove();
+					$("#confirmDeleteImageAlert").find('input').remove();
+					$("#confirmDeleteChoiceBlockAlert").find('input').remove();
+					$("#confirmDeleteSpaceAlert").hide();
```
indentation

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -991,6 +1181,107 @@
                     });
                 });
                 
+                function getSpaceById(spaceId, callback) {
```
get space by id should use an api call that actually gets the space by its id, not search all and then iterate over all!

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -991,6 +1181,107 @@
+                            let spaceName = null;
+                            for (let i = 0; i < result.length; i++) {
+                                const space = result[i];
+                                if (space.id === spaceId) {
```
where is `spaceId` defined?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -991,6 +1181,107 @@
+                    });
+                }
+                
+                function getCurrentSpace(blockId) {
```
why is this getting `blockId` but is called `getCurrentSpace` and what's the difference to `getSpaceById`?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -991,6 +1181,107 @@
+			            var spaceId = selectedSpace.options[selectedSpace.selectedIndex].value;
+			            formData.append('title', title);
+			            formData.append('spaceId', spaceId);
+			            getSpaceById(spaceId, function(spaceName){
```
for what does this need to be a callback?

jdamerow replied:
In fact, why does `getSpaceById` to be called here? The space block should only need the space and a title, which you have after selecting a space, don't you?

jdamerow commented on vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:
```diff
@@ -35,16 +35,25 @@
+            var currentSpaceId;
             var addVideoClk=0;
             var editVideoClk=0;
+            var currentBlockId;
```
I'm am fairly certain both `currentSpaceId` and `currentBlockId` do not need to be global variables.

ajay3568yadav replied:
without currentBlockId as a global variable close Space Block does not work as it is not able to get the id of the block, if you have any other suggestion then lmk.

jdamerow commented:
Make it so, Jenkins.
